pkg/ebpf/c/capture_filtering.h

```diff
@@ -93,15 +99,41 @@ statfunc bool filter_file_type(void *ctx, void *filter_map, size_t map_idx, stru
             if (pipe != NULL) {
                 type_filter_match = true;
             }
-        } else if (*type_filter & FILTER_SOCKET_FILES) {
+        }
+        if (*type_filter & FILTER_SOCKET_FILES) {
             if (imode_mode & S_IFSOCK) {
                 type_filter_match = true;
             }
-        } else if (*type_filter & FILTER_NORMAL_FILES) {
+        }
+        if (*type_filter & FILTER_NORMAL_FILES) {
             if (imode_mode & S_IFREG) {
                 type_filter_match = true;
             }
         }
+        if (*type_filter & FILTER_ELF_FILES) {
+            file_id_t file_id = get_file_id(file);
+            file_id.ctime = 0;
+            if (start_pos == 0) {
+                // Check if header is matching ELF header
+                u8 header[FILE_MAGIC_HDR_SIZE] = {};
+                fill_file_header(io_data, header);
+                if (header[0] == 0x7f && header[1] == 0x45 && header[2] == 0x4c &&
+                    header[3] == 0x46) {
+                    type_filter_match = true;
```

Comment:
nit: since you changed the `if...else if... else` pattern to be `if...;if...;` and it seems that once any of the filters matched you set type_filter_match to true (thus an OR filter), you can skip any other checks after a match and go directly to the return code (e.g. using goto)